lux/core/frame.py

```diff
@@ -168,12 +179,15 @@ def _infer_structure(self):
         # If the dataframe is very small and the index column is not a range index, then it is likely that this is an aggregated data
         is_multi_index_flag = self.index.nlevels != 1
         not_int_index_flag = not pd.api.types.is_integer_dtype(self.index)
-        small_df_flag = len(self) < 100
+        small_df_flag = len(self) < 100 and lux.config.executor.name == "PandasExecutor"
         if self.pre_aggregated == None:
             self.pre_aggregated = (is_multi_index_flag or not_int_index_flag) and small_df_flag
             if "Number of Records" in self.columns:
                 self.pre_aggregated = True
-            self.pre_aggregated = "groupby" in [event.name for event in self.history]
+            self.pre_aggregated = (
+                "groupby" in [event.name for event in self.history]
+                and lux.config.executor.name == "PandasExecutor"
+            )
 
```

Comment:
why do the checks have to be for `lux.config.executor.name`? Can these attributes not be computed by the SQLTable? If so, should we flip the order of the statements so it short-circuits?